Chess/desk.cpp

```diff
@@ -42,8 +45,123 @@ Desk::Desk(QObject *parent) :
     for (int i=0; i<8; i++) {
         buffer_[6][i] = new Pawn(this, fcolor);
     }
+}
+
+
+bool Desk::grabFigure(const Cell &cell)
+{
+    Figure *curr_fig = getFigure(cell);
+
+    if (curr_fig == NULL)
+        return false;
+
+    if (curr_fig->getColor() != color_to_move_)
+        return false;
+
+    have_active_figure_ = true;
+
+    command_ = new Command(this);
+    command_->set_b_info(cell);
+
+    return true;
+}

```

Comment:
I would put all this code into command::execute. grabbing figure, chacking it existsa etc...

In case it's connected with UI (i.e. called when user touches figure on UI) i wouldn't have it on this level. Only on UI controller